src/app.rs

```diff
@@ -1,27 +1,20 @@
-use crate::browser::dom::virtual_dom_bridge;
 use crate::browser::{
     service::routing,
     util::{self, window, ClosureNew},
     Url, DUMMY_BASE_URL,
 };
-use crate::virtual_dom::{patch, El, EventHandlerManager, IntoNodes, Mailbox, Node, Tag};
-use builder::{
-    init::{Init, InitFn as BuilderInitFn},
-    IntoAfterMount, MountPointInitInitAPI, UndefinedInitAPI, UndefinedMountPoint,
-};
-use enclose::{enc, enclose};
+use crate::virtual_dom::{patch, El, EventHandlerManager, IntoNodes, Mailbox, Tag};
+use enclose::enclose;
 use std::{
     any::Any,
     cell::{Cell, RefCell},
     collections::VecDeque,
-    marker::PhantomData,
     rc::Rc,
 };
-use types::{RoutesFn, SinkFn, UpdateFn, ViewFn, WindowEventsFn};
+use types::{UpdateFn, ViewFn};
```

Comment:
We should consider to refactor / remove `types` module.

Reply:
next PR, or add it to this one?

Reply:
I don't know how hard it would be - your choice.

Reply:
I'll leave it to another PR. Taking out `window_events` might reveal some bugs, and I we _might_ be biting off more than we can chew already. 

I'm happy for this PR to be put onto a dev-branch, then bring in a bunch of tests for the examples (and real-world app, I think) to make sure everything's okay. Then we can look at types. How does that sound?

Reply:
I think this PR should be merged into `master` before we'll create a new one. You don't want to resolve complicated conflicts once somebody push bigger changes to `master` and forgot context while you are working on other parts in a new PR.

Reply:
Hmmm, I can only disagree from a place of relative lack of experience. I can see the logic in it.